apm/src/app/user/user.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import { IUser } from './user';
+
+@Injectable()
+
+export class UserService {
+
+  public currentUser: IUser | null = null;
+    
+  public login(userName: string, password: string) {
+    this.currentUser = {
+      id : 1,
+      firstName: userName,
+      lastName: ''
+    }
+  }
+
+  public isAuthenticated() {
+    return Boolean(this.currentUser);
```

Comment:
since we have modified the type of currentUser, I think we need to check for it to be null ?

Reply:
like `return this.currentUser !== null`